Source/SwiftLintFramework/Rules/UnusedOptionalBindingRule.swift

```diff
@@ -40,23 +42,31 @@ public struct UnusedOptionalBindingRule: ASTRule, ConfigurationProviderRule {
             "if let (_, second) = getOptionalTuple(), let ↓_ = Foo.optionalValue {\n" +
             "}\n",
             "if let ↓(_, _, _) = getOptionalTuple(), let bar = Foo.optionalValue {\n" +
-            "}\n"
+            "}\n",
+            "func foo() {\nif let ↓_ = bar {\n}\n"
         ]
     )
 
     public func validate(file: File,
                          kind: StatementKind,
                          dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+        let conditionKind = "source.lang.swift.structure.elem.condition_expr"
         guard kind == .if || kind == .guard,
-            let offset = dictionary.offset,
-            let length = dictionary.length,
-            let range = file.contents.bridge().byteRangeToNSRange(start: offset, length: length) else {
+            let elements = dictionary.elements?.filter({ $0.kind == conditionKind }) else {
```

Comment:
this was needed to make sure that the pattern was in the condition, not inside an `if` body